lib/ansible/modules/database/postgresql/postgresql_query.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: postgresql_query
+short_description: Run arbitrary queries on a postgresql database.
+description:
+   - Run arbitrary SQL queries against a PostgreSQL instance from the current host.
+   - Query results are returned as "query_results" as a list of dictionaries.
+     Number of rows affected are returned as "row_count".
+   - Can read queries from a .sql script files.
+   - SQL scripts may be templated with ansible variables at execution time.
+version_added: "2.4"
+options:
+  db:
+    description:
+      - name of the database to run queries against.
+    required: false
+    default: postgres
+  port:
+    description:
+      - Database port to connect to.
+    required: false
+    default: 5432
+  login_user:
+    description:
+      - User (role) used to authenticate with PostgreSQL.
+    required: false
+    default: postgres
+  login_password:
+    description:
+      - Password used to authenticate with PostgreSQL.
+    required: false
+    default: null
+  login_host:
+    description:
+      - Host running PostgreSQL.
+    required: false
+    default: localhost
+  login_unix_socket:
+    description:
+      - Path to a Unix domain socket for local connections.
+    required: false
+    default: null
+  query:
+    description:
+      - SQL query to run. Variables can be escaped with psycopg2 syntax.
+      - Can be a sql scripts file.
+    required: true
+  positional_args:
+    description:
+      - A list of values to be passed as positional arguments to the query.
+      - Cannot be used with named_args.
+    required: false
+    default: null
+  autocommit:
+      description:
+        - Enable transaction autocommit.
+        - If enabled, This WILL run the query in check_mode.
```

Comment:
I don't love this as behavior you can just "turn on" since it's potentially very surprising. If this is enabled I think the module should stop supporting check_mode (or just lie and always report a change).